mathesar/tests/api/test_record_api.py

```diff
@@ -1527,6 +1527,7 @@ def test_record_post_exclusion_violation(reservations_table, engine, client):
         ("room_number" WITH =, TSRANGE("check_in_date", "check_out_date", '[]') WITH &&);"""
     )
     with engine.begin() as conn:
+        conn.execute(text("""SET search_path="Reservations";"""))
```

Comment:
@Anish9901 Why was this change necessary? Will this affect app behavior? I.e., do we now need to check that we're manually setting the search path when we previously weren't?

Reply:
The problem is that these tests would fail when running our entire test suite but would pass when run individually or with the `--lf` flag. I think the issue has to do with the way pytest handles its fixtures. This change was irrelevant to this PR however, the aforementioned issue existed since these tests were initially added so I decided to fix them in this PR. I have removed the `SET search_path` statement and fixed the underlying fixture that was causing the issue in the following commit:  https://github.com/mathesar-foundation/mathesar/pull/3478/commits/c3120967fb50933c3d20a58383c898ccee23fb2e.